pinot-minion/src/main/java/org/apache/pinot/minion/BaseMinionStarter.java

```diff
@@ -240,6 +240,16 @@ public String getStatusDescription() {
     LOGGER.info("Pinot minion started");
   }
 
+  private void updateInstanceConfigIfNeeded() {
+    InstanceConfig instanceConfig = HelixHelper.getInstanceConfig(_helixManager, _instanceId);
+    boolean updated = HelixHelper.updateHostnamePort(instanceConfig, _host, _port);
+    updated |= HelixHelper.addDefaultTags(instanceConfig,
```

Comment:
I thought about this but will the "short circuit" feature stop the function from calling in some cases? @Jackie-Jiang

Reply:
Short circuit won't apply to `|` but only to `||`